contracts/ERC677BridgeTokenRewardable.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity 0.4.24;
+
+import "./ERC677BridgeToken.sol";
+
+
+contract ERC677BridgeTokenRewardable is ERC677BridgeToken {
+
+    address public blockRewardContract;
+    address public validatorSetContract;
+
+    constructor(
+        string _name,
+        string _symbol,
+        uint8 _decimals
+    ) public ERC677BridgeToken(_name, _symbol, _decimals) {}
+
+    function setBlockRewardContract(address _blockRewardContract) onlyOwner public {
+        require(_blockRewardContract != address(0) && isContract(_blockRewardContract));
+        blockRewardContract = _blockRewardContract;
+    }
+
+    function setValidatorSetContract(address _validatorSetContract) onlyOwner public {
+        require(_validatorSetContract != address(0) && isContract(_validatorSetContract));
+        validatorSetContract = _validatorSetContract;
+    }
+
+    modifier onlyBlockRewardContract() {
+        require(msg.sender == blockRewardContract);
+        _;
+    }
+
+    modifier onlyValidatorSetContract() {
+        require(msg.sender == validatorSetContract);
+        _;
+    }
+
+    function mintReward(address[] _receivers, uint256[] _rewards) external onlyBlockRewardContract {
+        for (uint256 i = 0; i < _receivers.length; i++) {
+            address to = _receivers[i];
+            uint256 amount = _rewards[i];
+
+            // Mint `amount` for `to`
+            totalSupply_ = totalSupply_.add(amount);
+            balances[to] = balances[to].add(amount);
+            emit Mint(to, amount);
+            emit Transfer(address(0), to, amount);
+        }
+    }
+
+    function stake(address _staker, uint256 _amount) external onlyValidatorSetContract {
+        // Transfer `_amount` from `_staker` to `validatorSetContract`
+        require(_amount <= balances[_staker]);
+        balances[_staker] = balances[_staker].sub(_amount);
```

Comment:
may we use `super.transferFrom(_staker, validatorSetContract, _amount)` here?

Reply:
We can't use `transferFrom` because it requires allowance and uses `msg.sender`:

![image](https://user-images.githubusercontent.com/33550681/50452321-7e238e00-094a-11e9-9986-ef10d8f7b1ae.png)

We could use [`_transfer` internal function](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/7361ffd26b6354499d0b68d6d2d9da1f8cb21c8b/contracts/token/ERC20/ERC20.sol#L136) which is available in the last version of `openzeppelin-solidity` (npm package update is required for that).